src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -0,0 +1,498 @@
+                          this.selectedElementRect.top -
+                          this.topOfPageAdjustment -
+                          (this.currentTourStep.scrollAdjustment ? this.currentTourStep.scrollAdjustment : 0) +
+                          this.getStepScreenAdjustment()
+                        : 0;
+                    try {
+                        window.scrollTo({
+                            left: 0,
+                            top: topPos,
+                            behavior: 'smooth',
+                        });
+                    } catch (err) {
+                        if (err instanceof TypeError) {
+                            window.scroll(0, topPos);
+                        } else {
+                            throw err;
+                        }
+                    }
+                } else {
+                    // Scroll so the element is on the bottom of the screen.
+                    const topPos =
+                        this.selectedElementRect && this.currentTourStep
+                            ? window.scrollY +
+                              this.selectedElementRect.top +
+                              this.selectedElementRect.height -
+                              window.innerHeight +
+                              (this.currentTourStep.scrollAdjustment ? this.currentTourStep.scrollAdjustment : 0) -
+                              this.getStepScreenAdjustment()
+                            : 0;
+                    try {
+                        window.scrollTo({
+                            left: 0,
+                            top: topPos,
+                            behavior: 'smooth',
+                        });
+                    } catch (err) {
+                        if (err instanceof TypeError) {
+                            window.scroll(0, topPos);
+                        } else {
+                            throw err;
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * Handle orb event
+     */
+    public handleOrb(): void {
+        this.guidedTourService.activateOrb();
+        if (this.currentTourStep && this.currentTourStep.selector) {
+            this.scrollToAndSetElement();
+        }
+    }
+
+    /**
+     * Return if tour step is visible on screen
+     */
+    private isTourOnScreen(): boolean {
+        if (!this.currentTourStep && !this.currentTourStep!.selector) {
+            return false;
+        }
+        return this.tourStep && this.elementInViewport(document.querySelector(this.currentTourStep!.selector!)) && this.elementInViewport(this.tourStep.nativeElement);
+    }
+
+    /**
+     * Define if HTMLElement is visible in current viewport
+     * Modified from https://stackoverflow.com/questions/123999/how-to-tell-if-a-dom-element-is-visible-in-the-current-viewport
+     * @param element
+     */
+    private elementInViewport(element: HTMLElement | null): boolean {
```

Comment:
This could be a useful util and doesn't have to be specific to this component.

Reply:
@thilo-behnke That's true. I will implement this in another PR